src/components/NewSignIn/index.tsx

```diff
@@ -37,22 +33,35 @@ export default function NewSignInPage() {
   });
 
   useEffect(() => {
-    if (!router.isReady) return;
-    isQuery &&
-      (async () => {
-        try {
-          const { data } = await API.get(`/oauth/${router.query.client_id}`);
-          setServiceName(data.serviceName);
-        } catch (e) {
-          if (isAxiosError(e) && e.response!.status === 404) {
-            toast.error('해당하는 서비스가 없습니다.');
-            router.back();
-          } else {
-            toast.error('예기치 못한 오류가 발생하였습니다.');
-          }
+    const autoOauth = async () => {
+      const { data } = await API.post('/oauth/code/access');
+      if (isQuery)
+        router.replace(`${router.query.redirect_uri}?code=${data.code}`);
+    };
+
+    const getService = async () => {
+      try {
+        const { data } = await API.get(`/oauth/${router.query.client_id}`);
+        setServiceName(data.serviceName);
+        if (checkAuto) autoOauth();
+      } catch (e) {
+        if (isAxiosError(e) && e.response!.status === 404) {
+          toast.error('해당하는 서비스가 없습니다.');
+          router.back();
+        } else {
+          toast.error('예기치 못한 오류가 발생하였습니다.');
         }
-      })();
-  }, [router.isReady, isQuery, router]);
+      }
+    };
+
+    if (!router.isReady || checkAuto === undefined) return;
+    if (isQuery) {
+      getService();
+    } else {
+      checkAuto ? router.push('/') : '';
```

Comment:
```suggestion
      checkAuto && router.push('/'); 
```
별건 없긴 한데 useEffect에서 문자열 반환할 것은 아니니까 바꿔봤어